src/System.Slices/src/System/Span.cs

```diff
@@ -335,7 +320,7 @@ public override bool Equals(object obj)
         /// gain: performance: no boundaries check (single operation) 
         /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private T GetItemWithoutBoundariesCheck(int index)
+        internal T GetItemWithoutBoundariesCheck(int index)
```

Comment:
I think this is a valid scenario. If you name something dangerous then it is dangerous to use it and I wouldn't worry about bounds checks. What I see is that @adamsitnik can't implement extension on top of public surface on top of Span<T> because this is inefficient which sounds like a missing functionality on Span. If we can't implement it in our own library then no one can. Considering that span is acting as base for any data operations I call this should be public. Let's resolve this separately though. I think we should get rid of all "internal" in our libraries as they seem to be as pretty as "friend" in C++